src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy

```diff
@@ -21,11 +21,29 @@ class CourseInstallDecoderService implements CourseInstallDecoderPort {
     }
 
     @Override
-    void decodeCourseInstallEvent( String event, int schemaId ) {
+    void decodeCourseInstallV1ToV1( String event, int schemaId ) {
+        def schema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        log.info( 'The avro schema for event decoding : ' + schema.toString( true ) )
+
+        decoder.decodeEventWithOldSchema( schema, event )
```

Comment:
Yes. We should now with what version of schema the event was encoded. We get the id of the schema in the message headers and retrieve it from the schema registry.